File: LeetCode-Java/Sum Root to Leaf Numbers.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int sumNumbers(TreeNode root) {
        return sumNumbersPart(root, 0);
    }
    
    public int sumNumbersPart(TreeNode root, int tempSum)
    {
        if(root == null) return 0;
        tempSum = tempSum * 10 + root.val;
        if(root.left == null && root.right == null)
            return tempSum;
        else
            return sumNumbersPart(root.left, tempSum) + sumNumbersPart(root.right, tempSum);
    }
}

File: Leetcode-cpp/sum_root_to_leaf_numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int sumNumbers(TreeNode* root) {
        return sumNumbersHelper(root, 0);
    }
    int sumNumbersHelper(TreeNode* root, int pre) {
        if (root == nullptr) return 0;
        pre = 10 * pre + root->val;
        if (root->left == nullptr && root->right == nullptr) return pre;
        int result = 0;
        if (root->left != nullptr) result += sumNumbersHelper(root->left, pre);
        if (root->right != nullptr) result += sumNumbersHelper(root->right, pre);
        return result;
    }
};